saleor/discount/utils.py

```diff
@@ -430,17 +445,9 @@ def prepare_line_discount_objects_for_catalogue_promotions(
             line_discounts_to_remove.extend(discounts_to_update)
             continue
 
-        # delete the discount objects that are not valid anymore
-        line_discounts_to_remove.extend(
-            _get_discounts_that_are_not_valid_anymore(
-                line_info.rules_info,
-                rule_id_to_discount,  # type: ignore[arg-type]
-            )
-        )
-
-        for rule_info in line_info.rules_info:
+        if line_info.rules_info:
```

Comment:
What will happen in the case when:
I will have some discount assigned to `line_info` == `discount_to_update` is not None but `line_info.rules_info` will be empty list or None?  As far as I understand in that case, the catalogue discount should be removed. Do we do that? Is it event possible?